modules/system/models/File.php

```diff
@@ -80,7 +80,7 @@ public function getPublicPath()
             $uploadsPath .= '/protected';
         }
 
-        return Url::asset($uploadsPath) . '/';
+        return Storage::url($uploadsPath) . '/';
```

Comment:
Isn't the better option to actually do `$this->getDisk()->url($uploadsPath)` without changing the filesystems config? Not sure why you think that change is required but it's definitely not getting merged.

Reply:
@Teranode to clarify I meant the filesystems.php config change wasn't going to get merged. Still possibly open to the other change if it solves a specific problem.

Reply:
> @Teranode to clarify I meant the filesystems.php config change wasn't going to get merged. Still possibly open to the other change if it solves a specific problem.

I changed the filesystems.php file because when using the 'url' param with `Storage::url()` and 'app/storage' is defined within it, it appends `app/storage` twice, most likely because the `path` param from a storage config is what is being used when defining the base directory  (a safer bet wuld have been to use `env('APP_URL')`.

In any case i've already deleted my fork.